### DISCUSSION TITLE
Opus 1.5.1 - LACE and NoLACE

### DISCUSSION THREAD

**Discussion by szabiz** [enzo1982/freac#604] [Audio Format Discussion]
How to enable  LACE and NoLACE in the program of the Opus 1.5.1 codecs (6, 9, 12 kb/s) .

"To try LACE and NoLACE, just add the --enable-osce configure flag when building Opus. Then, to enable LACE at run-time, set the decoder complexity to 6. Set it to 7 or higher to enable NoLACE instead of LACE. "
https://opus-codec.org/demo/opus-1.5/

Thank you very much!

**Comment by enzo1982** [comment]
In recent [continuous builds](https://github.com/enzo1982/freac/releases/tag/continuous) of fre:ac, NoLACE is enabled by default.

You can control the decoder complexity in the configuration dialog under _Components->Decoders->Opus Audio Decoder_ to use LACE instead or completely disable LACE/NoLACE.

**Reply by szabiz** [reply]
Unfortunately, with the "Complexity" slider, I did not achieve the same sound quality as the Demo sounds on the Opus official website. So I don't see the blessings of version 1.5.1 with Lace and NoLace. Are you sure fre:ac can handle this feature?

**Reply by enzo1982** [reply]
Yes, fre:ac continuous builds definitely make use of LACE/NoLACE when decoding low-bitrate Opus. Try decoding this example file at q5 vs. q7 - you can clearly hear the difference.

[Sophies Welt.zip](https://github.com/enzo1982/freac/files/14796472/Sophies.Welt.zip)

However, this example uses a bitrate of 8 kbps which makes it a bit more difficult to hear the difference than the 6 kbps example on the Opus 1.5 demo page. But LACE/NoLACE is implemented in Opus only for 16 kHz internal SILK sample rate and I found that when encoding speech at 6 kbps, Opus/SILK usually switches to 12 kHz. Thus LACE/NoLACE will not be used for those files. They apparently used some tricks / very specific settings to create 16 kHz SILK encodings at 6 kbps for the demo page.
